api: include RepositoryPriority in FileMatch results.

rmmh opened the pull request:
This will allow the frontend to sort results from multiple backends by
repo priority, without having to do additional lookups for each result.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sourcegraph/zoekt/89)
<!-- Reviewable:end -->

rmmh reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/sourcegraph/zoekt/89)* status: 0 of 4 files reviewed, 2 unresolved discussions (waiting on @stefanhengl)

___
*[api.go, line 30 at r1](https://reviewable.io/reviews/sourcegraph/zoekt/89#-McKmeB2nnGZRJigth4b-r1-30:-McKmgeV7ihqJ-sgeko7:b-oxm6mq) ([raw file](https://github.com/sourcegraph/zoekt/blob/1f6a66fb29c52304a1bd2630196fc2620a4acbca/api.go#L30)):*
<details><summary><i>Previously, tsenart wrote…</i></summary><blockquote>

What's the reason for not using the `Score` field?
</blockquote></details>

Score is the quality of a match inside a repository, not a global repository ranking.


<!-- Sent from Reviewable.io -->

keegancsmith reviewed (commented):
We should just be able to use Score here iff we made it the same as our priority. We don't need a new field.

> *[Reviewable](https://reviewable.io/reviews/sourcegraph/zoekt/89#-:-McOdwN27kZ7QmV1gMnX:b-83y9zo)* status: 0 of 4 files reviewed, 2 unresolved discussions (waiting on @stefanhengl)

___
*[api.go, line 30 at r1](https://reviewable.io/reviews/sourcegraph/zoekt/89#-McKmeB2nnGZRJigth4b-r1-30:-McOdOrx9CsTagrpW7aN:b-tjb7ea) ([raw file](https://github.com/sourcegraph/zoekt/blob/1f6a66fb29c52304a1bd2630196fc2620a4acbca/api.go#L30)):*
<details><summary><i>Previously, rmmh (Ryan Hitchman) wrote…</i></summary><blockquote>

Score is the quality of a match inside a repository, not a global repository ranking.
</blockquote></details>

@rmmh zoekt uses score to globally rank results. We should just keep using it. We can adjust the heuristics that zoekt uses to create the final Score field for now to only use priority.


<!-- Sent from Reviewable.io -->

Commit:
api: add ShardPriority and MaxPendingShardPriority to SearchResult.

MaxPendingShardPriority gives precise information about what other
searches are inflight, in particular the largest outstanding priority
that hasn't completed yet.

This allows a receiver to know precise bounds on what priority results
can occur after a given search result, making it possible for it to
produce a well-ordered stream of search results from the
partially-ordered stream that zoekt produces.

rmmh commented:
Closing in favor of #104.
